core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -143,7 +192,38 @@ private void cacheManifests(FileIO fileIO) {
 
     if (allManifests == null) {
       // if manifests isn't set, then the snapshotFile is set and should be read to get the list
-      this.allManifests = ManifestLists.read(fileIO.newInputFile(manifestListLocation));
+      InputFile manifestListFile = fileIO.newInputFile(manifestListLocation);
+      if (manifestListKeyMetadata != null) { // encrypted manifest list file
+        Preconditions.checkArgument(
+            fileIO instanceof EncryptingFileIO,
+            "No encryption in FileIO class " + fileIO.getClass());
+        EncryptingFileIO encryptingFileIO = (EncryptingFileIO) fileIO;
+        Preconditions.checkArgument(
+            encryptingFileIO.encryptionManager() instanceof StandardEncryptionManager,
+            "Encryption manager for encrypted manifest list files can currently only be an instance of "
```

Comment:
Cannot decrypt manifest list (%s) because the encryption manager does not implement StandardEncryptionManager